htdocs/commande/list.php

```diff
@@ -973,7 +979,7 @@
 		if (! empty($arrayfields['s.nom']['checked']))
 		{
 			print '<td class="tdoverflowmax200">';
-			print $companystatic->getNomUrl(1, 'customer');
```

Comment:
This is why we call the var $companystatic: because we use it in a same way than a static object (so an object that is not initialized at each pass of the loop). So this is consume no database access and no memory (the same var in memory is erased by next loop).
After your change, you make a new for each different value. If there is 1000 different company on list for example and object Societe need 250 Ko , you will consume in memory 1000 x 250 Ko =  250 MB. With current code, you just consume 250 Ko. So you change is a good idea but is not well implemented. Don't forget that "memory only" manipulation of var must be considered as so fast that it has no effect on time compared to IO disk on SQL access.

The idea to use a cache to call getNomUrl is a good idea because inside the getNomUrl there is a call to "showphoto" and this method need to make IO on disk. So we can avoid this. This may also reduce number of logs too.

So this is what you must do:
- Keep object $companystatic = new Societe($db);
- Add your array $getNomUrl = array();   (but you can rename it   $cacheGetNomUrl so we know it's an array used for caching result of getNomUrl.
- Keep assignement of property into $companystatic and use it to fill the $cacheGetNomUrl.
- Instead of testing if (!isset($companystatic[$obj->socid])) {   just test:  if (isset($cacheGetNomUrl[$obj->socid]))

With that, we will have the cake and eat it too.

Reply:
Hi
Tks for this complete analysis and advice.